### PR TITLE
Fix a breaking change in get_puzzle_and_solution RPC

### PR DESCRIPTION
**Pull request by Quexington**

### CODE REVIEW

File: chia/rpc/full_node_rpc_api.py [hunk 1]
```diff
@@ -702,9 +701,10 @@ async def get_puzzle_and_solution(self, request: Dict) -> EndpointResult:
         if error is not None:
             raise ValueError(f"Error: {error}")
 
-        puzzle_ser: SerializedProgram = SerializedProgram.from_program(Program.to(puzzle))
-        solution_ser: SerializedProgram = SerializedProgram.from_program(Program.to(solution))
```
Comment (arvidn) [inline chia/rpc/full_node_rpc_api.py:706]: avoiding these conversions look like an optimization, does it affect correctness?

**Comment by arvidn** [comment]
@Quexington can you explain what the problem was?

File: chia/rpc/full_node_rpc_api.py (hunk 1 above)
Comment (arvidn) [reply chia/rpc/full_node_rpc_api.py:706]: `Program.to()` does not support converting from `SerializedProgram`

**Review by arvidn — APPROVED** [review]

**Comment by Quexington** [comment]
Explaining for github purposes: the RPC wasn't tested at all which is how this slipped through, but `get_puzzle_and_solution_for_coin` switched its API to return `SerializedProgram` which was causing an error when passed to `Program.to()`

**Review by trepca — APPROVED** [review]

**Pull request merged**